[Benchmark] Support CVQA

timothycdc opened the pull request:
Adds support for the [Cultural VQA (CVQA)](https://arxiv.org/pdf/2406.05967) benchmark.
Despite 'VQA' being in the name, CVQA is actually multiple-choice. 

<img width="662" height="246" alt="Screenshot 2025-07-25 at 12 01 12 AM" src="https://github.com/user-attachments/assets/b639007a-391d-46f9-aacc-62ab94d7ddeb" />


I used the dataset's region+language column (e.g. Japan, Japanese) as `l2-category split`.


[Original dataset link](https://huggingface.co/datasets/afaji/cvqa)
[TSV converted version](https://huggingface.co/datasets/timothycdc/VLMEvalKit_CVQA_GT/blob/main/CVQA.tsv)

timothycdc committed:
Add support for CVQA Benchmark

timothycdc committed:
Attempt 1: Fix linting

FangXinyu-0913 commented:
HI @timothycdc, Thanks for your contribution to our community. Here are the results of the test on our side for your reference. If you feel okay, we will merge this PR.

---------------------------------  -------------------
split                              test
Overall                            0.3877964141122036
('Amharic', 'Ethiopia')            0.24786324786324787
('Bengali', 'India')               0.18181818181818182
('Breton', 'France')               0.2962962962962963
('Bulgarian', 'Bulgaria')          0.3692722371967655
('Chinese', 'China')               0.4212218649517685
('Chinese', 'Singapore')           0.3915094339622642
('Egyptian_Arabic', 'Egypt')       0.1724137931034483
('Filipino', 'Philippines')        0.4482758620689655
('Hindi', 'India')                 0.3283582089552239
('Igbo', 'Nigeria')                0.325
('Indonesian', 'Indonesia')        0.4029126213592233
('Irish', 'Ireland')               0.4110429447852761
('Japanese', 'Japan')              0.3103448275862069
('Javanese', 'Indonesia')          0.3602693602693603
('Kinyarwanda', 'Rwanda')          0.33617021276595743
('Korean', 'South Korea')          0.3724137931034483
('Malay', 'Malaysia')              0.473015873015873
('Marathi', 'India')               0.24257425742574257
('Minangkabau', 'Indonesia')       0.350597609561753
('Mongolian', 'Mongolia')          0.22115384615384615
('Norwegian', 'Norway')            0.5418060200668896
('Oromo', 'Ethiopia')              0.38317757009345793
('Portuguese', 'Brazil')           0.5985915492957746
('Romanian', 'Romania')            0.5198675496688742
('Russian', 'Russia')              0.55
('Sinhala', 'Sri_Lanka')           0.21777777777777776
('Spanish', 'Argentina')           0.5471698113207547
('Spanish', 'Chile')               0.5897435897435898
('Spanish', 'Colombia')            0.5311203319502075
('Spanish', 'Ecuador')             0.5552486187845304
('Spanish', 'Mexico')              0.4613003095975232
('Spanish', 'Spain')               0.660377358490566
('Spanish', 'Uruguay')             0.44126984126984126
('Sundanese', 'Indonesia')         0.35
('Swahili', 'Kenya')               0.37362637362637363
('Tamil', 'India')                 0.18691588785046728
('Telugu', 'India')                0.27
('Urdu', 'India')                  0.2
('Urdu', 'Pakistan')               0.10648148148148148
Brands / products / companies      0.3910355486862442
Cooking and food                   0.35877862595419846
Geography / buildings / landmarks  0.4
Objects / materials / clothing     0.3337531486146096
People and everyday life           0.4195624195624196
Plants and animal                  0.36418816388467373
Public Figure and pop culture      0.41467576791808874
Sports and recreation              0.41372141372141374
Traditions / art / history         0.39231738035264485
Vehicles and Transportation        0.4074074074074074
---------------------------------  -------------------

BTW, the SYSPROMPT in the PR is not used in the actual evaluation. If you want it to be used by all models, you can write a `build_prompt` function in combination with this prompt and put it under the `class CVQA`.

FangXinyu-0913 commented:
> Thanks @FangXinyu-0913. Which model did you use for this? The score seems a bit lower than normal.
> 
> I will double-check the accuracy code and fix the `SYSPROMPT`.

I use the llava_v1.5_7b model and get this result.

kennymckormick commented:
BTW, a large proportion of VLMs may override the build_prompt method for MCQ benchmarks. So if you would like to force all VLMs to use the official prompt, you may consider change the DATASET_TYPE to CVQA_MCQ, etc.

> HI @timothycdc, Thanks for your contribution to our community. Here are the results of the test on our side for your reference. If you feel okay, we will merge this PR.
> 
> split test Overall 0.3877964141122036 ('Amharic', 'Ethiopia') 0.24786324786324787 ('Bengali', 'India') 0.18181818181818182 ('Breton', 'France') 0.2962962962962963 ('Bulgarian', 'Bulgaria') 0.3692722371967655 ('Chinese', 'China') 0.4212218649517685 ('Chinese', 'Singapore') 0.3915094339622642 ('Egyptian_Arabic', 'Egypt') 0.1724137931034483 ('Filipino', 'Philippines') 0.4482758620689655 ('Hindi', 'India') 0.3283582089552239 ('Igbo', 'Nigeria') 0.325 ('Indonesian', 'Indonesia') 0.4029126213592233 ('Irish', 'Ireland') 0.4110429447852761 ('Japanese', 'Japan') 0.3103448275862069 ('Javanese', 'Indonesia') 0.3602693602693603 ('Kinyarwanda', 'Rwanda') 0.33617021276595743 ('Korean', 'South Korea') 0.3724137931034483 ('Malay', 'Malaysia') 0.473015873015873 ('Marathi', 'India') 0.24257425742574257 ('Minangkabau', 'Indonesia') 0.350597609561753 ('Mongolian', 'Mongolia') 0.22115384615384615 ('Norwegian', 'Norway') 0.5418060200668896 ('Oromo', 'Ethiopia') 0.38317757009345793 ('Portuguese', 'Brazil') 0.5985915492957746 ('Romanian', 'Romania') 0.5198675496688742 ('Russian', 'Russia') 0.55 ('Sinhala', 'Sri_Lanka') 0.21777777777777776 ('Spanish', 'Argentina') 0.5471698113207547 ('Spanish', 'Chile') 0.5897435897435898 ('Spanish', 'Colombia') 0.5311203319502075 ('Spanish', 'Ecuador') 0.5552486187845304 ('Spanish', 'Mexico') 0.4613003095975232 ('Spanish', 'Spain') 0.660377358490566 ('Spanish', 'Uruguay') 0.44126984126984126 ('Sundanese', 'Indonesia') 0.35 ('Swahili', 'Kenya') 0.37362637362637363 ('Tamil', 'India') 0.18691588785046728 ('Telugu', 'India') 0.27 ('Urdu', 'India') 0.2 ('Urdu', 'Pakistan') 0.10648148148148148 Brands / products / companies 0.3910355486862442 Cooking and food 0.35877862595419846 Geography / buildings / landmarks 0.4 Objects / materials / clothing 0.3337531486146096 People and everyday life 0.4195624195624196 Plants and animal 0.36418816388467373 Public Figure and pop culture 0.41467576791808874 Sports and recreation 0.41372141372141374 Traditions / art / history 0.39231738035264485 Vehicles and Transportation 0.4074074074074074
> 
> BTW, the SYSPROMPT in the PR is not used in the actual evaluation. If you want it to be used by all models, you can write a `build_prompt` function in combination with this prompt and put it under the `class CVQA`.

timothycdc committed:
CVQA support both EN and LOC versions + custom prompt

timothycdc committed:
Merge branch 'main' into cvqa_PR

timothycdc commented:
Hi, thanks for your advice- I have updated the PR.

It turns out that there are two versions of CVQA, which I have now implemented:
1.   `CVQA_LOC`, which is the original benchmark I introduced. 
  -  It has multicultural questions, each in their local language. 
  - It is meant for VLMs trained with many languages.
2. (New)`CVQA_EN`, which is the benchmark which has multicultural questions translated to English. 
  -  It is supposed to be for VLMs trained primarily on English. 
  - This is why llava_v1.5_7b scored lower than normal, it was supposed to be evaluated on `CVQA_EN`, not `CVQA_LOC`, which was the first dataset that was tested.

I also put the prompt into `build_prompt` under CVQA. 

I will test this on my end first, then will update you when it's ready to merge

timothycdc committed:
Fix __init__.py syntax error
